GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -72,10 +92,25 @@ private bool TryInitializeUpgrader(out string error)
                     jsonTracer.AddLogFileEventListener(logFilePath, EventLevel.Informational, Keywords.Any);
 
                     this.tracer = jsonTracer;
-                    this.prerunChecker = new InstallerPreRunChecker(this.tracer, this.Confirmed ? GVFSConstants.UpgradeVerbMessages.GVFSUpgradeConfirm : GVFSConstants.UpgradeVerbMessages.GVFSUpgrade);
+                    string commandToRerun;
+
+                    if (this.Confirmed)
+                    {
+                        commandToRerun = GVFSConstants.UpgradeVerbMessages.GVFSUpgradeConfirm;
+                    }
+                    else if (this.DryRun)
+                    {
+                        commandToRerun = GVFSConstants.UpgradeVerbMessages.GVFSUpgradeDryRun;
+                    }
+                    else
+                    {
+                        commandToRerun = GVFSConstants.UpgradeVerbMessages.GVFSUpgrade;
+                    }
+
+                    this.prerunChecker = new InstallerPreRunChecker(this.tracer, commandToRerun);

```

Comment:
This seems pretty complicated to let the user know to re-run with --confirm. Do you think the distinction between --dry-run and --confirm is important here? Should we simplify by having the `InstallerPreRunChecker` always assume you want to end up with --confirm?

Reply:
agree... reverted.